File: fake_classification/fk_classifiers.py
import numpy as np
import pandas as pd
import time
from sklearn.model_selection import train_test_split
from sklearn.ensemble import RandomForestClassifier
from sklearn.naive_bayes import GaussianNB

df = pd.read_csv('data/data_500000s_128f.csv')

X = df.iloc[:, :-1]
y = df.iloc[:, -1]
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3, random_state=42)

#rf = RandomForestClassifier(n_estimators=10, random_state=42)  
nb = GaussianNB()

#Train RF
time_train = []
for i in range(0,1):
    start = time.time()
    rf.fit(X_train, y_train)  
    end = time.time()
    time_train.append(end - start)
np.mean(time_train)

#Train NB
time_train = []
for i in range(0,11):
    start = time.time()
    nb.fit(X_train, y_train)  
    end = time.time()
    time_train.append(end - start)
np.mean(time_train)

#Predictions (always with 100k datasets)
df = pd.read_csv('data/data_100000s_128f.csv')

X = df.iloc[:, :-1]
y = df.iloc[:, -1]
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3, random_state=42)

#rf = RandomForestClassifier(n_estimators=10, random_state=42)  
nb = GaussianNB()
nb.fit(X_train, y_train)  

#Predict
time_pred = []
for i in range(0,11):
    start = time.time()
    y_pred = nb.predict(X_test.iloc[0:30000, :])
    end = time.time()
    time_pred.append(end - start)
np.mean(time_pred)

File: fake_classification/fk_classifiers.jl
using CSV
using DataFrames
using MLDataUtils
using DecisionTree
using NaiveBayes
import Statistics: mean

data = CSV.read("data/data_500000s_128f.csv")

train, test = splitobs(shuffleobs(data), at = 0.7)
features = convert(Matrix, dropmissing(train)[:, 1:end-1])
labels = convert(Array, dropmissing(train)[:, end])

#Train RF
time_train = Float32[]
for i = 1:10
    append!(time_train, @elapsed model = build_forest(labels, features))
end
println(mean(time_train))

#Train NaiveBayes
time_train = Float32[]
model = GaussianNB(unique(labels), size(features)[2])
features = convert(Matrix, features')
for i = 1:10
    append!(time_train, @elapsed fit(model, features, labels))
end
println(mean(time_train))

#Predictions (always with 100k datasets)
data = CSV.read("data/data_100000s_128f.csv")
train, test = splitobs(shuffleobs(data), at = 0.7)
features = convert(Matrix, dropmissing(train)[:, 1:end-1])
labels = convert(Array, dropmissing(train)[:, end])

#model = build_forest(labels, features)
model = GaussianNB(unique(labels), size(features)[2])
features = convert(Matrix, features')
fit(model, features, labels)

time_train = Float32[]
obs = convert(Array, test[1:30000, 1:end-1])
obs = Array(obs')
for i = 1:10
    #append!(time_train, @elapsed apply_forest(model, obs))
    append!(time_train, @elapsed NaiveBayes.predict(model, obs))
end
println(mean(time_train))
